core/src/main/scala/com/databricks/labs/remorph/intermediate/adf/activities/LibraryDefinition.scala

```diff
@@ -0,0 +1,28 @@
+package com.databricks.labs.remorph.intermediate.adf.activities
+
+import com.databricks.labs.remorph.intermediate.adf.PipelineNode
+
+case class LibraryDefinition(
+    jarFilePath: Option[String],
+    eggFilePath: Option[String],
+    whlFilePath: Option[String],
+    mavenSpecification: Option[MavenSpecification],
+    pyPiSpecification: Option[PyPiSpecification],
+    cranSpecification: Option[CranSpecification])
```

Comment:
Following the comment above, here it is quite clear that we have three mutually exclusive cases mashed together. 
We should rather define it as
```scala
sealed trait LibraryDefinition
case class MavenLibrary(jarFilePath: String, mavenSpecification: MavenSpecification) extends LibraryDefinition
case class PyPiLibrary(eggFilePath: String, piPySpecification: PyPiSpecification) extends LibraryDefinition
case class CranLibrary(whlFilePath: String, cranSpecification: CranSpecification) extends LibraryDefinition
```